packages/bsky/src/api/app/bsky/notification/listNotifications.ts

```diff
@@ -93,24 +94,45 @@ const paginateNotifications = async (opts: {
   }
 }
 
+/**
+ * Applies a configurable delay to the datetime string of a cursor,
+ * effectively allowing for a delay on listing the notifications.
+ * This is useful to allow time for services to process notifications
+ * before they are listed to the user.
+ */
+export const delayCursor = (
+  cursorStr: string | undefined,
+  delayMs: number,
+): string => {
+  const nowMinusDelay = Date.now() - delayMs
+  if (cursorStr === undefined) {
+    return new Date(nowMinusDelay).toISOString()
+  }
+  const cursor = new Date(cursorStr).getTime()
```

Comment:
I'd propose we pass the value through rather than manipulate it if it does not parse as a date.